src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -379,6 +478,11 @@ CHIP_ERROR ChipDnssdRemoveServices()
         return CHIP_NO_ERROR;
     }
 
+    err = MdnsContexts::GetInstance().RemoveAllOfType(ContextType::RegisterRecord);
+    if (CHIP_ERROR_KEY_NOT_FOUND == err)
+    {
+        return CHIP_NO_ERROR;
+    }
```

Comment:
You probably want to update this code a little bit. Let's say there is no `ContextType::Register` the code before this changes is returning early a `CHIP_NO_ERROR`. But we may want to run the code to remove the `ContextType::RegisterRecord too`

So maybe something like:
```
    auto err = MdnsContexts::GetInstance().RemoveAllOfType(ContextType::Register);
    if (CHIP_ERROR_KEY_NOT_FOUND == err)
    {
         err = CHIP_NO_ERROR;
    }
    ReturnErrorOnFailure(err);

    err = MdnsContexts::GetInstance().RemoveAllOfType(ContextType::RegisterRecord);
    if (CHIP_ERROR_KEY_NOT_FOUND == err)
    {
         err = CHIP_NO_ERROR;
    }
    ReturnErrorOnFailure(err);

    return err;
```

Reply:
i already did this. Maybe you missed it https://github.com/project-chip/connectedhomeip/blob/726c61d5c8441d57de010820d65ab06fd6a27f90/src/platform/Darwin/DnssdImpl.cpp#L481

Reply:
done.